### PR TITLE
[4] Add intl extension check to system info, and while there clean up deprecations/types/[]

### PR DESCRIPTION
**Pull request by PhilETaylor**
### Summary of Changes

As a result of https://github.com/joomla/joomla-cms/pull/33451 and https://github.com/joomla/joomla-cms/pull/33454 I noticed ICONV was checked but not INTL - as both are needed for `File::makeSafe` it makes sense to have both listed in the System Information for ease of reference. 

While in this code, took the opportunity to remove deprecated `getDBO` method calls and add param types and return types. 

Screenshot 2021-05-01 at 00 18 43

### Testing Instructions

Load  System Information in Joomla Admin - if it loads, great. Check PHP tab and see your status of INTL (checks for presence of the function `transliterator_transliterate` which is the one Joomla uses in `File::makeSafe`

Try JSON and TXT download.

### Actual result BEFORE applying this Pull Request

Works - but intl extension not in the list

### Expected result AFTER applying this Pull Request

Still works - but calls one less deprecated function and code is slightly more modern and less warnings in phpStorm. 

### Documentation Changes Required

None

### CODE REVIEW

File: administrator/language/en-GB/com_admin.ini [hunk 1]
```diff
@@ -119,6 +119,7 @@ COM_ADMIN_HELP_USERS_USER_MANAGER_EDIT="Users: New/Edit"
 COM_ADMIN_HELP_USERS_USER_NOTES="Users: User Notes"
 COM_ADMIN_HELP_USERS_USER_NOTES_EDIT="Users: User Notes - New/Edit"
 COM_ADMIN_ICONV_AVAILABLE="Iconv Available"
+COM_ADMIN_INTL_AVAILABLE="Intl Available"
```
Comment (brianteeman) [inline administrator/language/en-GB/com_admin.ini]: it would be so much better if "intl" was hardcoded. It shouldn't be translated

Comment (PhilETaylor) [reply administrator/language/en-GB/com_admin.ini]: I just did the same as Iconv :)

Comment (brianteeman) [reply administrator/language/en-GB/com_admin.ini]: Doesn't make it right

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 68d52d8174900012aa70528e65d5ae9460fba604<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33457">issues.joomla.org/tracker/joomla-cms/33457</a>.</sub>

File: administrator/components/com_admin/src/Model/SysinfoModel.php [hunk 2]
```diff
@@ -32,23 +32,23 @@ class SysinfoModel extends BaseDatabaseModel
...
+	protected $config = [];
 
 	/**
 	 * Some system values
 	 *
 	 * @var    array
 	 * @since  1.6
 	 */
-	protected $info = array();
+	protected $info = [];
```
Comment (sandewt) [inline administrator/components/com_admin/src/Model/SysinfoModel.php:51]: See also line 68

`protected $phpInfoArray;`

should be

`protected $phpInfoArray = [];`

Comment (sandewt) [reply administrator/components/com_admin/src/Model/SysinfoModel.php:51]: Line 150

`protected $safeData;`

should be

`protected $safeData = [];`

**Comment by richard67** [comment]
~~If this PR will be merged, the same check should also be added to the Joomla Update Component's Pre-Update Checks. This will have to be done in the 3.10-dev branch so it can be checked before updating from 3.10 to 4. I could do that if necessary.~~

~~@PhilETaylor Do I understand right that the INTL extension is only needed in J4, not in J3? Or is that wrong and it should also be checked in J3? If my understanding is right, then the future PR for the pre-update check should show the check for INTL only if the version to be updated to is J4, like we have it already with the database compatibility check for MS SQL Server.~~

Update: I just see the INTL is only needed for transliteration, so it makes sense to add it to the system info with this PR here, but it is _not a required setting_ to be checked by the Pre-Update Check. What could make sense is to add it to the _recommended_ settings in the Pre-Update Checker.

File: administrator/components/com_admin/src/Model/SysinfoModel.php (hunk 2 above)
Comment (richard67) [reply administrator/components/com_admin/src/Model/SysinfoModel.php:51]: @sandewt I don't agree. This PR here changes `array()` to short array syntax `[]`. The changes which you suggest mean to change the initial value from unset/null to an empty array. If that is right or not has to be checked elsewhere but seems to me to be out of scope of this PR and has to be carefully checked.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 68d52d8174900012aa70528e65d5ae9460fba604<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33457">issues.joomla.org/tracker/joomla-cms/33457</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33457">issues.joomla.org/tracker/joomla-cms/33457</a>.</sub>

**Comment by richard67** [comment]
Drone failure is **_not_** related to this PR just by the way.

File: administrator/components/com_admin/src/Model/SysinfoModel.php (hunk 2 above)
Comment (sandewt) [reply administrator/components/com_admin/src/Model/SysinfoModel.php:51]: @richard, indeed you're right.

**Comment by richard67** [comment]
Restarted drone. All fine now.

**Comment by Quy** [comment]
Thank you!

**Pull request merged**